src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp

```diff
@@ -52,6 +76,32 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
             output_dims[i] = query_dims[permute_axes[i]];
         }
 
+        if (input_shapes.size() == 7 && !m_config.is_causal) {
```

Comment:
May we add a lambda function and call it twice to reuse the similar logic?

Reply:
fix, when permute_axes.empty() is empty, I make permute_axes = {0 ,1 ,2, 3}, so it make the code simpler